src/nni_manager/training_service/kubernetes/kubeflow/kubeflowConfig.ts

```diff
@@ -0,0 +1,179 @@
+import { MethodNotImplementedError } from '../../../common/errors';
+
+/** operator types that kubeflow supported */
+export type KubeflowOperator = 'tf-operator' | 'pytorch-operator' ;
+export type DistTrainRole = 'worker' | 'ps' | 'master';
+export type KubeflowJobStatus = 'Created' | 'Running' | 'Failed' | 'Succeeded';
+export type OperatorApiVersion = 'v1alpha2' | 'v1beta1';
+
+export class KubeflowClusterConfig extends KubernetesClusterConfig {
+    public readonly operator: KubeflowOperator;
+    constructor(codeDir: string, operator: KubeflowOperator) {
+        super(codeDir);
+        this.operator = operator;
+    }
+}
+
+export class KubeflowClusterConfigNFS extends KubernetesClusterConfigNFS {
+    public readonly operator: KubeflowOperator;
+    constructor(
+            operator: KubeflowOperator, 
+            apiVersion: string, 
+            nfs: NFSConfig,
+            storage?: KubernetesStorageKind
+        ) {
+        super(apiVersion, nfs, storage);
+        this.operator = operator;
+    }
+
+    public get storageType(): KubernetesStorageKind {
+        return 'nfs';
+    }
+
+    public static getInstance(jsonObject: object): KubeflowClusterConfigNFS {
+        let kubeflowClusterConfigObjectNFS = <KubeflowClusterConfigNFS>jsonObject;
+        return new KubeflowClusterConfigNFS(
+            kubeflowClusterConfigObjectNFS.operator,
+            kubeflowClusterConfigObjectNFS.apiVersion,
+            kubeflowClusterConfigObjectNFS.nfs,
+            kubeflowClusterConfigObjectNFS.storage
+        );
+    }
+}
+
+export class KubeflowClusterConfigAzure extends KubernetesClusterConfigAzure{
+    public readonly operator: KubeflowOperator;
+    
+    constructor(
+            operator: KubeflowOperator, 
+            apiVersion: string, 
+            keyVault: keyVaultConfig, 
+            azureStorage: AzureStorage, 
+            storage?: KubernetesStorageKind
+        ) {
+        super(apiVersion, keyVault, azureStorage,storage);
+        this.operator = operator;
+    }
+
+    public get storageType(): KubernetesStorageKind{
+        return 'azureStorage';
+    }
+
+    public static getInstance(jsonObject: object): KubeflowClusterConfigAzure {
+        let kubeflowClusterConfigObjectAzure = <KubeflowClusterConfigAzure>jsonObject;
+        return new KubeflowClusterConfigAzure(
+            kubeflowClusterConfigObjectAzure.operator,
+            kubeflowClusterConfigObjectAzure.apiVersion,
+            kubeflowClusterConfigObjectAzure.keyVault,
+            kubeflowClusterConfigObjectAzure.azureStorage,
+            kubeflowClusterConfigObjectAzure.storage
+        );
+    }
+}
+
+export class KubeflowClusterConfigFactory {
+
+    public static generateKubeflowClusterConfig(jsonObject: object): KubeflowClusterConfig {
+         let storageConfig = <StorageConfig>jsonObject;
+         if(storageConfig.storage && storageConfig.storage === 'azureStorage') {
```

Comment:
Suggest to add the following statements after you convert jsonObject to StorageConfig:
if(!storageConfig) {
   throw new Error("Invalid json object as a StorageConfig instance");
}

Reply:
resolved.